digitalocean/resource_digitalocean_droplet_snapshot.go

```diff
@@ -0,0 +1,163 @@
+package digitalocean
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/digitalocean/godo"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceDigitalOceanDropletSnapshot() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDigitalOceanDropletSnapshotCreate,
+		Read:   resourceDigitalOceanDropletSnapshotRead,
+		Delete: resourceDigitalOceanDropletSnapshotDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"droplet_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"regions": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"size": {
+				Type:     schema.TypeFloat,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"min_disk_size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceDigitalOceanDropletSnapshotCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*godo.Client)
+
+	resourceId, _ := strconv.Atoi(d.Get("droplet_id").(string))
+	action, _, err := client.DropletActions.Snapshot(context.Background(), resourceId, d.Get("name").(string))
+	if err != nil {
+		return fmt.Errorf("Error creating Droplet Snapshot: %s", err)
+	}
+
+	if err = waitForAction(client, action); err != nil {
+		return fmt.Errorf(
+			"Error waiting for Droplet snapshot (%v) to finish: %s", resourceId, err)
+	}
+
+	snapshotList, err := SnapshotList(context.Background(), client)
+
+	if err != nil {
+		return fmt.Errorf("Error retriving Droplet Snapshot: %s", err)
+	}
+
+	for _, v := range snapshotList {
+		createdTime, _ := time.Parse("2006-01-02T15:04:05Z", v.Created)
+		checkTime := godo.Timestamp{createdTime}
```

Comment:
```suggestion
		checkTime := godo.Timestamp{Time: createdTime}
```
This will make the `go vet` happy.

Reply:
done